src/manager.rs

```diff
@@ -0,0 +1,194 @@
+use compile::ExecutableDevEnv;
+use parse::Parser;
+use run::{ Execution, ExecutionEvent, Error as RunError };
+use rusqlite;
+use serde_json::error::Error as SerdeJsonError;
+use std::collections::HashMap;
+use std::io::Error as IOError;
+use std::path::{ Path, PathBuf };
+use std::sync::mpsc::RecvError;
+use transformable_channels::mpsc::{ channel, ExtSender };
+
+
+/// A ScriptManager error.
+#[derive(Debug)]
+pub enum Error {
+    NoSuchScriptError,
+    ScriptAlreadyRunning,
+    RusqliteError(rusqlite::Error),
+    IoError(IOError),
+    ExecutionError(ExecutionEvent),
+    RunError(RunError),
+    JsonError(SerdeJsonError),
+    RecvError(RecvError)
+}
+
+/// ScriptManager stores a persistent database of scripts and executes them.
+/// Each script can be individually enabled or disabled.
+/// When a script is enabled, it is always running (unless an error occured during launch).
+pub struct ScriptManager<Env> where Env: ExecutableDevEnv + 'static {
+    path: PathBuf,
+    env: Env,
+    runners: HashMap<String, Execution<Env>>
+}
+
+impl<Env> ScriptManager<Env> where Env: ExecutableDevEnv + 'static {
+
+    /// Create a ScriptManager using a SQLite database file at the given path.
+    /// If the database file does not exist, it will be created.
+    pub fn new(env: Env, path: &Path) -> Result<Self, Error> {
+        let connection = try!(rusqlite::Connection::open(&path));
+        try!(connection.execute("CREATE TABLE IF NOT EXISTS scripts (
+            name        TEXT NOT NULL PRIMARY KEY,
+            source      TEXT NOT NULL,
```

Comment:
Could you document whether the source has already been parsed before being stored?